[bug] v1.11.0 native Windows gem causes nokogumbo installation to fail

**Please describe the bug**

Nokogumbo cannot build against the Nokogiri v1.11.0 native Windows gem.

It is able to build against
- the v1.10.10 native Windows gem
- the v1.11.0 native Linux and Darwin gems.

**Help us reproduce what you're seeing**

See the Nokogumbo test suite's [failing windows jobs when built against v1.11.0](https://github.com/rubys/nokogumbo/actions/runs/461489890):

```
MAKE='make -j2 V=1' bundle exec rake -- --with-libxml2
mkdir -p tmp/x64-mingw32/nokogumbo/2.7.2
cd tmp/x64-mingw32/nokogumbo/2.7.2
C:/hostedtoolcache/windows/Ruby/2.7.2/x64/bin/ruby.exe -I. ../../../../ext/nokogumbo/extconf.rb -- --with-libxml2
checking for whether -ID:/a/nokogumbo/nokogumbo/vendor/bundle/ruby/2.7.0/gems/nokogiri-1.11.0-x64-mingw32/ext/nokogiri is accepted as CFLAGS... yes
checking for whether -ID:/a/nokogumbo/nokogumbo/vendor/bundle/ruby/2.7.0/gems/nokogiri-1.11.0-x64-mingw32/ext/nokogiri/include is accepted as CFLAGS... yes
checking for whether -ID:/a/nokogumbo/nokogumbo/vendor/bundle/ruby/2.7.0/gems/nokogiri-1.11.0-x64-mingw32/ext/nokogiri/include/libxml2 is accepted as CFLAGS... yes
checking for libxml/tree.h... yes
checking for nokogiri.h... yes
creating Makefile
cd -
cd tmp/x64-mingw32/nokogumbo/2.7.2
make -j2 V=1
(echo EXPORTS && echo Init_nokogumbo) > nokogumbo-x64-mingw32.def
...
x86_64-w64-mingw32-gcc -shared -o nokogumbo.so nokogumbo.o ascii.o attribute.o char_ref.o error.o foreign_attrs.o parser.o string_buffer.o string_piece.o svg_attrs.o svg_tags.o tag.o tag_lookup.o tokenizer.o token_buffer.o utf8.o util.o vector.o -L. -L/C/hostedtoolcache/windows/Ruby/2.7.2/x64/lib -L. -pipe -s -fstack-protector-strong -pipe -s -fstack-protector-strong -Wl,--enable-auto-image-base,--enable-auto-import nokogumbo-x64-mingw32.def    -lx64-msvcrt-ruby270  -lshell32 -lws2_32 -liphlpapi -limagehlp -lshlwapi  
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: nokogumbo.o:nokogumbo.c:(.text+0x34): undefined reference to `xmlFreeDoc'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: nokogumbo.o:nokogumbo.c:(.text+0x10a8): undefined reference to `xmlNewDocText'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: nokogumbo.o:nokogumbo.c:(.text+0x10c4): undefined reference to `xmlAddChild'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: nokogumbo.o:nokogumbo.c:(.text+0x10dd): undefined reference to `xmlNewDocNode'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: nokogumbo.o:nokogumbo.c:(.text+0x111e): undefined reference to `xmlAddChild'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: nokogumbo.o:nokogumbo.c:(.text+0x1153): undefined reference to `xmlNewNsProp'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: nokogumbo.o:nokogumbo.c:(.text+0x1191): undefined reference to `xmlSearchNs'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: nokogumbo.o:nokogumbo.c:(.text+0x11af): undefined reference to `xmlNewNs'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: nokogumbo.o:nokogumbo.c:(.text+0x11c8): undefined reference to `xmlNewDocComment'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: nokogumbo.o:nokogumbo.c:(.text+0x120d): undefined reference to `xmlNewCDataBlock'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: nokogumbo.o:nokogumbo.c:(.text+0x123e): undefined reference to `xmlSearchNs'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: nokogumbo.o:nokogumbo.c:(.text+0x1260): undefined reference to `xmlNewNs'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: nokogumbo.o:nokogumbo.c:(.text+0x127e): undefined reference to `xmlSearchNs'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: nokogumbo.o:nokogumbo.c:(.text+0x12a0): undefined reference to `xmlNewNs'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: nokogumbo.o:nokogumbo.c:(.text+0x12e6): undefined reference to `xmlSearchNs'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: nokogumbo.o:nokogumbo.c:(.text+0x12f6): undefined reference to `xmlSetNs'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: nokogumbo.o:nokogumbo.c:(.text+0x130e): undefined reference to `xmlSearchNs'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: nokogumbo.o:nokogumbo.c:(.text+0x132c): undefined reference to `xmlNewNs'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: nokogumbo.o:nokogumbo.c:(.text+0x1352): undefined reference to `xmlNewNs'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: nokogumbo.o:nokogumbo.c:(.text+0x1440): undefined reference to `htmlNewDocNoDtD'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: nokogumbo.o:nokogumbo.c:(.text+0x1462): undefined reference to `xmlCreateIntSubset'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: nokogumbo.o:nokogumbo.c:(.text+0x1484): undefined reference to `Nokogiri_wrap_xml_document'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: nokogumbo.o:nokogumbo.c:(.text+0x14c5): undefined reference to `htmlNewDocNoDtD'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: nokogumbo.o:nokogumbo.c:(.rdata$.refptr.mNokogiri[.refptr.mNokogiri]+0x0): undefined reference to `mNokogiri'
C:/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/10.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: nokogumbo.o:nokogumbo.c:(.rdata$.refptr.cNokogiriXmlSyntaxError[.refptr.cNokogiriXmlSyntaxError]+0x0): undefined reference to `cNokogiriXmlSyntaxError'
collect2.exe: error: ld returned 1 exit status
make: *** [Makefile:261: nokogumbo.so] Error 1
```

And the same suite's [passing status when built against v1.10.10](https://github.com/rubys/nokogumbo/actions/runs/461879340).


**Expected behavior**

Nokogumbo should be able to build identically against v1.11.0 as it does against v1.10.10.


**Environment**

From Nokogumbo's `ci.yml` config:

```
jobs:
  test:
    strategy:
      fail-fast: false
      matrix:
        os: [ubuntu, macos, windows]
        ruby: [2.1, 2.2, 2.3, 2.4, 2.5, 2.6, 2.7]
        system_libraries: [true, false]
        exclude:
        - os: macos
          ruby: 2.1
        - os: macos
          ruby: 2.2
        - os: macos
          ruby: 2.3
        - os: windows
          ruby: 2.1
        - os: windows
          ruby: 2.2
        - os: windows
          ruby: 2.3
    runs-on: ${{ matrix.os }}-latest

    steps:
    - uses: actions/checkout@v2
    - name: Set up Ruby
      uses: ruby/setup-ruby@v1
      with:
        ruby-version: ${{ matrix.ruby }}
        bundler-cache: true
        
    - name: install html5lib tests
      run: git clone --depth 1 --branch all-error-fixes --single-branch https://github.com/stevecheckoway/html5lib-tests.git test/html5lib-tests

    - name: Run tests
      env:
        USE_SYSTEM_LIBRARIES: ${{ matrix.system_libraries }}
      shell: bash
      run: ./scripts/ci-test.sh
```


**Additional context**

Note that Linux and Darwin native gems aren't a problem.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -11,6 +11,7 @@ Nokogiri follows [Semantic Versioning](https://semver.org/), please see the [REA
 * [CRuby] `NodeSet` may now safely contain `Node` objects from multiple documents. Previously the GC lifecycle of the parent `Document` objects could lead to contained nodes being GCed while still in scope. [[#1952](https://github.com/sparklemotion/nokogiri/issues/1952)]
 * [CRuby] Patch libxml2 to avoid "huge input lookup" errors on large CDATA elements. (See upstream [GNOME/libxml2#200](https://gitlab.gnome.org/GNOME/libxml2/-/issues/200) and [GNOME/libxml2!100](https://gitlab.gnome.org/GNOME/libxml2/-/merge_requests/100).) [[#2132](https://github.com/sparklemotion/nokogiri/issues/2132)].
 * [CRuby] `{XML,HTML}::Document.parse` now invokes `#initialize` exactly once. Previously `#initialize` was invoked twice on each object.
+* [CRuby+Windows] Enable Nokogumbo (and other downstream gems) to compile and link against `nokogiri.so` by including `LDFLAGS` in `Nokogiri::VERSION_INFO`. [[#2167](https://github.com/sparklemotion/nokogiri/issues/2167)]
 * [JRuby] `{XML,HTML}::Document.parse` now invokes `#initialize` exactly once. Previously `#initialize` was not called, which was a problem for subclassing such as done by `Loofah`.
 
 

--- a/ext/nokogiri/extconf.rb
+++ b/ext/nokogiri/extconf.rb
@@ -741,6 +741,10 @@ cross_build_p) do |recipe|
       recipe.configure_options += ["RANLIB=/usr/bin/ranlib", "AR=/usr/bin/ar"]
     end
 
+    if windows?
+      cflags = concat_flags(cflags, "-ULIBXML_STATIC", "-DIN_LIBXML")
+    end
+
     recipe.configure_options += [
       "--without-python",
       "--without-readline",

--- a/ext/nokogiri/extconf.rb
+++ b/ext/nokogiri/extconf.rb
@@ -239,7 +239,7 @@ def ensure_package_configuration(opt: nil, pc: nil, lib:, func:, headers:)
 end
 
 def ensure_func(func, headers = nil)
-  have_func(func, headers) || abort_could_not_find_library(lib)
+  have_func(func, headers) || abort_could_not_find_library(func)
 end
 
 def preserving_globals

--- a/ext/nokogiri/nokogiri.h
+++ b/ext/nokogiri/nokogiri.h
@@ -15,6 +15,14 @@
 #  include <windows.h>
 #endif
 
+#if _WIN32
+#  define NOKOPUBFUN __declspec(dllexport)
+#  define NOKOPUBVAR __declspec(dllexport) extern
+#else
+#  define NOKOPUBFUN
+#  define NOKOPUBVAR extern
+#endif
+
 
 #include <stdlib.h>
 #include <string.h>
@@ -83,47 +91,47 @@ xmlNodePtr xmlLastElementChild(xmlNodePtr parent);
 #endif
 
 
-extern VALUE mNokogiri ;
-extern VALUE mNokogiriHtml ;
-extern VALUE mNokogiriHtmlSax ;
-extern VALUE mNokogiriXml ;
-extern VALUE mNokogiriXmlSax ;
-extern VALUE mNokogiriXslt ;
+NOKOPUBVAR VALUE mNokogiri ;
+NOKOPUBVAR VALUE mNokogiriHtml ;
+NOKOPUBVAR VALUE mNokogiriHtmlSax ;
+NOKOPUBVAR VALUE mNokogiriXml ;
+NOKOPUBVAR VALUE mNokogiriXmlSax ;
+NOKOPUBVAR VALUE mNokogiriXslt ;
 
-extern VALUE cNokogiriSyntaxError;
-extern VALUE cNokogiriXmlAttr;
-extern VALUE cNokogiriXmlAttributeDecl;
-extern VALUE cNokogiriXmlCData;
-extern VALUE cNokogiriXmlCharacterData;
-extern VALUE cNokogiriXmlComment;
-extern VALUE cNokogiriXmlDocument ;
-extern VALUE cNokogiriXmlDocumentFragment;
-extern VALUE cNokogiriXmlDtd;
-extern VALUE cNokogiriXmlElement ;
-extern VALUE cNokogiriXmlElementContent;
-extern VALUE cNokogiriXmlElementDecl;
-extern VALUE cNokogiriXmlEntityDecl;
-extern VALUE cNokogiriXmlEntityReference;
-extern VALUE cNokogiriXmlNamespace ;
-extern VALUE cNokogiriXmlNode ;
-extern VALUE cNokogiriXmlNodeSet ;
-extern VALUE cNokogiriXmlProcessingInstruction;
-extern VALUE cNokogiriXmlReader;
-extern VALUE cNokogiriXmlRelaxNG;
-extern VALUE cNokogiriXmlSaxParser ;
-extern VALUE cNokogiriXmlSaxParserContext;
-extern VALUE cNokogiriXmlSaxPushParser ;
-extern VALUE cNokogiriXmlSchema;
-extern VALUE cNokogiriXmlSyntaxError;
-extern VALUE cNokogiriXmlText ;
-extern VALUE cNokogiriXmlXpathContext;
-extern VALUE cNokogiriXmlXpathSyntaxError;
-extern VALUE cNokogiriXsltStylesheet ;
+NOKOPUBVAR VALUE cNokogiriSyntaxError;
+NOKOPUBVAR VALUE cNokogiriXmlAttr;
+NOKOPUBVAR VALUE cNokogiriXmlAttributeDecl;
+NOKOPUBVAR VALUE cNokogiriXmlCData;
+NOKOPUBVAR VALUE cNokogiriXmlCharacterData;
+NOKOPUBVAR VALUE cNokogiriXmlComment;
+NOKOPUBVAR VALUE cNokogiriXmlDocument ;
+NOKOPUBVAR VALUE cNokogiriXmlDocumentFragment;
+NOKOPUBVAR VALUE cNokogiriXmlDtd;
+NOKOPUBVAR VALUE cNokogiriXmlElement ;
+NOKOPUBVAR VALUE cNokogiriXmlElementContent;
+NOKOPUBVAR VALUE cNokogiriXmlElementDecl;
+NOKOPUBVAR VALUE cNokogiriXmlEntityDecl;
+NOKOPUBVAR VALUE cNokogiriXmlEntityReference;
+NOKOPUBVAR VALUE cNokogiriXmlNamespace ;
+NOKOPUBVAR VALUE cNokogiriXmlNode ;
+NOKOPUBVAR VALUE cNokogiriXmlNodeSet ;
+NOKOPUBVAR VALUE cNokogiriXmlProcessingInstruction;
+NOKOPUBVAR VALUE cNokogiriXmlReader;
+NOKOPUBVAR VALUE cNokogiriXmlRelaxNG;
+NOKOPUBVAR VALUE cNokogiriXmlSaxParser ;
+NOKOPUBVAR VALUE cNokogiriXmlSaxParserContext;
+NOKOPUBVAR VALUE cNokogiriXmlSaxPushParser ;
+NOKOPUBVAR VALUE cNokogiriXmlSchema;
+NOKOPUBVAR VALUE cNokogiriXmlSyntaxError;
+NOKOPUBVAR VALUE cNokogiriXmlText ;
+NOKOPUBVAR VALUE cNokogiriXmlXpathContext;
+NOKOPUBVAR VALUE cNokogiriXmlXpathSyntaxError;
+NOKOPUBVAR VALUE cNokogiriXsltStylesheet ;
 
-extern VALUE cNokogiriHtmlDocument ;
-extern VALUE cNokogiriHtmlSaxPushParser ;
-extern VALUE cNokogiriHtmlElementDescription ;
-extern VALUE cNokogiriHtmlSaxParserContext;
+NOKOPUBVAR VALUE cNokogiriHtmlDocument ;
+NOKOPUBVAR VALUE cNokogiriHtmlSaxPushParser ;
+NOKOPUBVAR VALUE cNokogiriHtmlElementDescription ;
+NOKOPUBVAR VALUE cNokogiriHtmlSaxParserContext;
 
 typedef struct _nokogiriTuple {
   VALUE         doc;
@@ -169,7 +177,7 @@ VALUE noko_xml_node_set_wrap(xmlNodeSetPtr node_set, VALUE document) ;
 
 VALUE noko_xml_document_wrap_with_init_args(VALUE klass, xmlDocPtr doc, int argc, VALUE *argv);
 VALUE noko_xml_document_wrap(VALUE klass, xmlDocPtr doc);
-VALUE Nokogiri_wrap_xml_document(VALUE klass, xmlDocPtr doc); /* deprecated. use noko_xml_document_wrap() instead. */
+NOKOPUBFUN VALUE Nokogiri_wrap_xml_document(VALUE klass, xmlDocPtr doc); /* deprecated. use noko_xml_document_wrap() instead. */
 
 #define DOC_RUBY_OBJECT_TEST(x) ((nokogiriTuplePtr)(x->_private))
 #define DOC_RUBY_OBJECT(x) (((nokogiriTuplePtr)(x->_private))->doc)

--- a/lib/nokogiri/version/info.rb
+++ b/lib/nokogiri/version/info.rb
@@ -89,6 +89,11 @@ module Nokogiri
           nokogiri["version"] = Nokogiri::VERSION
 
           unless jruby?
+            #  enable gems like nokogumbo to build with the following in their extconf.rb:
+            #
+            #    append_cflags(Nokogiri::VERSION_INFO["nokogiri"]["cppflags"])
+            #    append_ldflags(Nokogiri::VERSION_INFO["nokogiri"]["ldflags"])
+            #
             cppflags = ["-I#{header_directory.shellescape}"]
             ldflags = []
 

--- a/lib/nokogiri/version/info.rb
+++ b/lib/nokogiri/version/info.rb
@@ -10,6 +10,14 @@ module Nokogiri
       ::JRUBY_VERSION if ::RUBY_PLATFORM == "java"
     end
 
+    def windows?
+      ::RUBY_PLATFORM =~ /mingw|mswin/
+    end
+
+    def ruby_minor
+      Gem::Version.new(::RUBY_VERSION).segments[0..1].join(".")
+    end
+
     def engine
       defined?(::RUBY_ENGINE) ? ::RUBY_ENGINE : "mri"
     end
@@ -74,6 +82,7 @@ module Nokogiri
 
     def to_hash
       header_directory = File.expand_path(File.join(File.dirname(__FILE__), "../../../ext/nokogiri"))
+
       {}.tap do |vi|
         vi["warnings"] = []
         vi["nokogiri"] = {}.tap do |nokogiri|
@@ -81,11 +90,25 @@ module Nokogiri
 
           unless jruby?
             cppflags = ["-I#{header_directory.shellescape}"]
+            ldflags = []
+
             if libxml2_using_packaged?
               cppflags << "-I#{File.join(header_directory, 'include').shellescape}"
               cppflags << "-I#{File.join(header_directory, 'include/libxml2').shellescape}"
+
+              if windows?
+                # on windows, nokogumbo needs to link against nokogiri.so to resolve symbols. see #2167
+                lib_directory = File.expand_path(File.join(File.dirname(__FILE__), "../#{ruby_minor}"))
+                unless File.exist?(lib_directory)
+                  lib_directory = File.expand_path(File.join(File.dirname(__FILE__), ".."))
+                end
+                ldflags << "-L#{lib_directory.shellescape}"
+                ldflags << "-l:nokogiri.so"
+              end
             end
+
             nokogiri["cppflags"] = cppflags
+            nokogiri["ldflags"] = ldflags
           end
         end
         vi["ruby"] = {}.tap do |ruby|

--- a/scripts/test-gem-installation
+++ b/scripts/test-gem-installation
@@ -43,6 +43,8 @@ Minitest::Reporters.use!([Minitest::Reporters::SpecReporter.new])
 
 puts "Testing #{gemspec.full_name} installed in #{gemspec.base_dir}"
 describe gemspec.full_name do
+  let(:ruby_maj_min) { Gem::Version.new(::RUBY_VERSION).segments[0..1].join(".") }
+  let(:nokogiri_lib_dir) { File.join(gemspec.gem_dir, "lib/nokogiri") }
   let(:nokogiri_ext_dir) { File.join(gemspec.gem_dir, "ext/nokogiri") }
   let(:nokogiri_include_dir) { File.join(nokogiri_ext_dir, "include") }
 
@@ -106,6 +108,20 @@ describe gemspec.full_name do
               found = true if File.file?(File.join(header_dir, header))
             end
             assert(found, "expected to find #{header} in one of: #{headers_dirs.inspect}")
+          end
+        end
+
+        it "has ldflags pointing to the shared object file" do
+          ldflags = Nokogiri::VERSION_INFO["nokogiri"]["ldflags"]
+          if ::RUBY_PLATFORM =~ /mingw|mswin/
+            if gemspec.platform.cpu
+              assert_includes(ldflags, "-L#{File.join(nokogiri_lib_dir, ruby_maj_min)}")
+            else
+              assert_includes(ldflags, "-L#{nokogiri_lib_dir}")
+            end
+            assert_includes(ldflags, "-l:nokogiri.so")
+          else
+            assert_empty(ldflags)
           end
         end
       end if Nokogiri::VersionInfo.instance.libxml2_using_packaged?

--- a/test/test_version.rb
+++ b/test/test_version.rb
@@ -18,8 +18,9 @@ module TestVersionInfoTests
     if jruby?
       refute(Nokogiri::VERSION_INFO["nokogiri"].has_key?("cppflags"), "did not expect cppflags")
     else
-      # cppflags are more fully tested in scripts/test-gem-installation
-      assert_kind_of(Array, Nokogiri::VERSION_INFO["nokogiri"]["cppflags"], "expected cppflags to be an array")
+      # cppflags/ldflags are more fully tested in scripts/test-gem-installation
+      assert_kind_of(Array, Nokogiri::VERSION_INFO["nokogiri"]["cppflags"], "cppflags should be an array")
+      assert_kind_of(Array, Nokogiri::VERSION_INFO["nokogiri"]["ldflags"], "ldflags should be an array")
     end
 
     assert_equal(::RUBY_VERSION, Nokogiri::VERSION_INFO["ruby"]["version"])

